lib/kernel/src/inet_res.erl

```diff
@@ -304,6 +305,12 @@ make_options([{verbose,Val}|Opts]=Opts0, [verbose|Names]=Names0) ->
        true ->
 	    erlang:error(badarg, [Opts0,Names0])
     end;
+make_options([{nxdomain_reply,Val}|Opts]=Opts0, [nxdomain|Names]=Names0) ->
+    if is_boolean(Val) ->
+	    [Val|make_options(Opts, Names)];
+       true ->
+	    erlang:error(badarg, [Opts0,Names0])
+    end;
```

Comment:
I think these two clauses should be merged like this, to avoid misspelling of a `Name` occurence:
``` erlang
make_options([{Name,Val}|Opts]=Opts0, [Name|Names]=Names0)
  when Name =:= verbose;
       Name =:= nxdomain_reply ->
    if is_boolean(Val) ->
        [Val|make_options(Opts, Names)];
```

Reply:
That misspeling is probably what caused the Dialyzer error